### PR TITLE
feat: Connect Frames to external pages

### LINKED ISSUE #63: Connect Frames to external pages
Everyone wants to link their Frames to custom webpages, so when the Frame itself is clicked in a client it navigates to their own link. Right now, the standard white "Hello from FrameTrain" page is displayed. 

To implement this, one would have to:
- Add a new column to the `frames` table, `linkedPage`. This is done by adding the field to the schema and running the `drizzle-kit` command in `package.json` to create a new migration, which is then applied using the relevant command.
- Add a `Popover` with an `Input` in the Editor's nav bar, that allows the user to input a URL.
- Inside the 2 `route` functions inside `/f`, get the value of `linkedPage` and feed it to `buildFramePage`.
- Inside `buildFramePage`, modify the returned html and add a `<script />` tag that redirects the user to the `linkedPage`. This means FrameTrain page will load and then redirect to the link set by the user.

### PR DESCRIPTION
**Pull request by waptik**
- Allow `Users` to attach an external link to their individual Frames
- Viewers upon clicking on the frame in any client, get redirect to the external page if set or to the standard white "Hello from FrameTrain" page

https://loom.com/share/78d64fc0c38e4c749830ddd8af38bd34

closes #63

### CODE REVIEW

**Comment by FTCHD** [comment]
1. Rename `upsert` to `update` in the server action name. You're technically correct but we strive for consistent naming. Rename the debounced callback as well.
2. Convert the args from an object to positional arguments, like `updateFrameLinkedPage(id, linkedPage)`.
3. Use `linkedPage: string | undefined` inside `lib/serve`.

**Comment by waptik** [comment]
> 1. Rename `upsert` to `update` in the server action name. You're technically correct but we strive for consistent naming. Rename the debounced callback as well.
> 2. Convert the args from an object to positional arguments, like `updateFrameLinkedPage(id, linkedPage)`.
> 3. Use `linkedPage: string | undefined` inside `lib/serve`.

Done!

File: lib/serve.ts [hunk 1]
```diff
@@ -23,10 +23,12 @@ export async function buildFramePage({
     component,
     image,
     functionName,
+    linkedPage,
 }: {
     id: string
+    linkedPage?: string
```
Comment (FTCHD) [inline lib/serve.ts:29]: This is where `linkedPage: string | undefined` should've been used.

File: app/(app)/f/[frameId]/route.ts [hunk 2]
```diff
@@ -26,7 +30,11 @@ export async function GET(request: Request, { params }: { params: { frameId: str
 
     const buildParameters = await initial(frame.config, frame.state)
 
-    const { frame: renderedFrame } = await buildFramePage({ id: frame.id, ...buildParameters })
+    const { frame: renderedFrame } = await buildFramePage({
+        id: frame.id,
+        linkedPage: frame.linkedPage ?? undefined,
```
Comment (FTCHD) [inline app/(app)/f/[frameId]/route.ts:35]: Not here, it was okay to keep it as `frame.linkedPage`.

File: app/(app)/f/[frameId]/[functionName]/[[...query]]/route.ts [hunk 3]
```diff
@@ -88,7 +96,8 @@ export async function POST(
     // no need to pass it back and forth in the future
     const { frame: renderedFrame, state: newState } = await buildFramePage({
         id: frame.id,
-        ...(buildParameters as FrameData),
+        linkedPage: frame.linkedPage ?? undefined,
```
Comment (FTCHD) [inline app/(app)/f/[frameId]/[functionName]/[[...query]]/route.ts:99]: or here

File: app/(app)/f/[frameId]/route.ts (hunk 2 above)
Comment (waptik) [reply app/(app)/f/[frameId]/route.ts:35]: `frame.linkedPage` is either string or null so i had to do it that way

File: lib/serve.ts (hunk 1 above)
Comment (waptik) [reply lib/serve.ts:29]: `linkedPage?: string` is still `linkedPage: string | undefined`. Or you explicitly want it to be `linkedPage: string | undefined`?

File: app/(app)/f/[frameId]/[functionName]/[[...query]]/route.ts (hunk 3 above)
Comment (waptik) [reply app/(app)/f/[frameId]/[functionName]/[[...query]]/route.ts:99]: i could make `linkedPage: string | null` inside `buildFramePage` to get rid of `?? undefined`, if that's ok!

**Pull request merged**